### PR TITLE
[13.0][MIG] report_qweb_signer

### PR DESCRIPTION
**Pull request by Laurent-Corron**

### CODE REVIEW

File: bi_sql_editor/models/bi_sql_view_field.py [hunk 1]
```diff
@@ -237,8 +237,8 @@ def _prepare_search_filter_field(self):
         self.ensure_one()
         res = ""
         if self.field_description and self.is_group_by:
-            res = """<filter name="%s" string="%s"
-                        context="{'group_by':'%s'}"/>""" % (
+            res = """<filter name="{}" string="{}"
+                        context="{{'group_by':'{}'}}"/>""".format(
```
Comment (sbidoul) [inline bi_sql_editor/models/bi_sql_view_field.py:241]: This change should not be in this PR ?

File: report_qweb_signer/data/defaults.xml [hunk 2]
```diff
@@ -2,6 +2,6 @@
 <odoo>
     <record model="ir.config_parameter" id="report_qweb_signer_java_param">
         <field name="key">report_qweb_signer.java_parameters</field>
-        <field name="value">-Xms4M -Xmx4M</field>
+        <field name="value">-Xms4M -Xmx1g -XX:CompressedClassSpaceSize=256m</field>
```
Comment (sbidoul) [inline report_qweb_signer/data/defaults.xml]: Is this change related to the migration ?

**Review by sbidoul — COMMENTED** [review]
Looks good.

File: bi_sql_editor/models/bi_sql_view_field.py (hunk 1 above)
Comment (Laurent-Corron) [reply bi_sql_editor/models/bi_sql_view_field.py:241]: This was changed automatically by pre-commit, I can remove it if you would like ?

File: report_qweb_signer/data/defaults.xml (hunk 2 above)
Comment (Laurent-Corron) [reply report_qweb_signer/data/defaults.xml]: If I didn't add -XX:CompressedClassSpaceSize=256m to the file, I couldn't run tests (same thing for a gitlab runner)
-Xmx1g was just a test I didn't remove, changed it back to -Xmx4M

File: bi_sql_editor/models/bi_sql_view_field.py (hunk 1 above)
Comment (sbidoul) [reply bi_sql_editor/models/bi_sql_view_field.py:241]: Ah ok, it must be pyupgrade that does it. Strange that it changes in this PR and while the main branch is green ?

**Comment by sbidoul** [comment]
@pedrobaeza if you are interested. There is also https://github.com/OCA/reporting-engine/pull/528 which adds optional endesive support (no java \o/).

**Comment by pedrobaeza** [comment]
Thanks for the pointer.

If the Java-free solution is working, shouldn't we go directly to that one and don't make it optional?

**Comment by sbidoul** [comment]
We could make endesive the default in 13.0+, and keep the java code path for a little while until people confirm they are happy with endesive.

**Comment by pedrobaeza** [comment]
If there's no drawback, I think the goal of the module is very clear: to sign the PDF. Letting 2 options for doing exactly the same is maintenance-wise? If there are some differences, then let's put them on the table for deciding.

**Comment by sbidoul** [comment]
I agree we'll need to keep only one at some point. But for now we can't promise that endesive has no drawbacks. It is simpler to deploy and has better performance, but there might be some exotic signature feature missing (such as the LTV option).

**Review by pedrobaeza — APPROVED** [review]
OK then, let's go with such approach.

Code review

**Review by sbidoul — APPROVED** [review]
